src/app/icd/server/ICDManager.cpp

```diff
@@ -188,15 +192,15 @@ void ICDManager::SendCheckInMsgs()
                 continue;
             }
 
-            if (entry.clientType == ClientTypeEnum::kEphemeral)
+            if (!ShouldCheckInMsgsBeSentAtActiveModeFunction(entry.fabricIndex, entry.monitoredSubject))
             {
-                // If the registered client is ephemeral, do not send a Check-In message
-                // continue to next entry
                 continue;
             }
 
-            if (!ShouldCheckInMsgsBeSentAtActiveModeFunction(entry.fabricIndex, entry.monitoredSubject))
+            // Validate Check-In BackOff strategy if we should send a Check-In message.
```

Comment:
I am unsure if this comment adds to reading the code below. Is there anything we can describe in comments that is not obvious from reading the if condition?

Reply:
Removed the comment since we don't what the underlying what the underlying Check-In BackOff strategy is here.